src/psyclone/psyir/tools/dependency_tools.py

```diff
@@ -401,7 +401,8 @@ def is_scalar_parallelisable(self, var_info):
         # Now we have at least two accesses. If the first access is a WRITE,
         # then the variable is not used in a reduction. This relies on sorting
         # the accesses by location.
-        if all_accesses[0].access_type == AccessType.WRITE:
+        if all_accesses[0].access_type in (AccessType.WRITE,
+                                           AccessType.READWRITE):
```

Comment:
I'm not sure about this. Doesn't a READWRITE imply a reduction? e.g.

    sum = 0.0
    do ji = 1,jpi
      sum = sum + var(ji)
    end do

where I assume this routine would be given the list of Accesses for the loop body? Perhaps @hiker can help here? Or do the accesses to `sum` on the LHS and RHS count as two different accesses?

Reply:
They count as different accesses. The function argument var_info considers them together, but we use this  var_info.all_accesses which considers them one by one. And in this condition we just access the all_accesses[0] (which I assume sorts the rhs to take place before).